internal/app/api/balance/handler.go

```diff
@@ -0,0 +1,168 @@
+}
+
+func (h *Handler) list(w http.ResponseWriter, r *http.Request) {
+	userID, ok := util.UserIDFromContext(r.Context())
+	if !ok {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	withdrawals, err := h.uc.UserWithdrawals(r.Context(), userID)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if len(withdrawals) < 1 {
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+
+	resp := make([]*listResponse, 0, len(withdrawals))
+	for _, withdrawal := range withdrawals {
+		resp = append(resp, &listResponse{
+			Order:       withdrawal.OrderNumber,
+			Sum:         withdrawal.Sum,
+			ProcessedAt: withdrawal.CreatedAt.Format(time.RFC3339),
+		})
+	}
+
+	b, err := json.Marshal(resp)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("content-type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	_, _ = w.Write(b)
+}
+
+func (h *Handler) balance(w http.ResponseWriter, r *http.Request) {
+	userID, ok := util.UserIDFromContext(r.Context())
+	if !ok {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	sumWC, err := h.uc.GetUserSumWC(r.Context(), userID)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	resp := &balanceResponse{
+		Current: sumWC.Current,
+		Sum:     sumWC.Withdrawn,
+	}
+
+	b, err := json.Marshal(resp)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	h.log.Info(
+		"user balance",
+		zap.String("response", string(b)),
+	)
+
+	w.Header().Set("content-type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	_, _ = w.Write(b)
+}
+
+func (h *Handler) withdraw(w http.ResponseWriter, r *http.Request) {
+	userID, ok := util.UserIDFromContext(r.Context())
+	if !ok {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	req := &withdrawRequest{}
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if err = json.Unmarshal(body, &req); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	b, err := h.uc.Withdraw(r.Context(), userID, req.Order, req.Sum)
+	if err != nil {
+		if errors.Is(err, domain.ErrorNumberValidateFormat) {
+			w.WriteHeader(http.StatusUnprocessableEntity)
+			return
+		} else if errors.Is(err, domain.ErrBalanceInsufficientFunds) {
+			w.WriteHeader(http.StatusPaymentRequired)
+			return
+		}
+
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	h.log.Info(
+		"balance withdraw",
+		zap.Float64("sum", b.Sum),
+		zap.Int("type", int(b.Type)),
+	)
+
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) RegisterRoutes(loader util.AuthSession) func(mux *chi.Mux) {
+	return func(mux *chi.Mux) {
+		mux.With(util.CheckAuth(loader)).Get("/api/user/withdrawals", h.list)
+		mux.With(util.CheckAuth(loader)).Get("/api/user/balance", h.balance)
+		mux.With(util.CheckAuth(loader)).Post("/api/user/balance/withdraw", h.withdraw)
```

Comment:
Можно сократить кол-во вызовов `With(...)`:
```suggestion
              routes := mux.With(util.CheckAuth(loader))
              routes.Get("/api/user/withdrawals", h.list)
              routes.Get("/api/user/balance", h.balance)
              routes.Post("/api/user/balance/withdraw", h.withdraw)
```